sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_secret.hpp

```diff
@@ -50,11 +50,28 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Secrets {
      */
     KeyVaultSecret(std::string name, std::string value)
         : Name(std::move(name)), Value(std::move(value))
+    {
+      if (Name.empty())
+      {
+        throw std::invalid_argument("Name cannot be empty");
+      }
+
+      if (Value.empty())
+      {
+        throw std::invalid_argument("Value cannot be empty");
+      }
+    };
+
+  private:
+    KeyVaultSecret(std::string name) : Name(std::move(name))
     {
       if (Name.empty())
       {
         throw std::invalid_argument("Name cannot be empty");
       }
     }
+
+    friend struct KeyVaultDeletedSecret;
```

Comment:
I want to know what @RickWinter will say here :)  .  He's something against the friend-strategy